src/main/java/org/opensearch/sdk/BaseExtension.java

```diff
@@ -35,10 +36,40 @@ public abstract class BaseExtension implements Extension {
     protected ThreadPool threadPool;
 
     /**
-     * Empty constructor to fulfill abstract class requirements
+     * Optional classpath-relative path to a yml file containing extension settings.
+     */
+    protected static final String EXTENSION_SETTINGS_PATH = "/sample/helloworld-settings.yml";
+
+    /**
+     * The extension settings include a name, host address, and port.
+     */
+    protected ExtensionSettings settings;
```

Comment:
You'll be setting this in the constructor, so make it `private final`.  And add a getter (or move it up from the subclass) that overrides `getExtensionSettings()`.

Reply:
> @dbwiddis Why do we need add `final` for `ExtensionSettings settings`?

It's not so much a "need" as good programming practice if we know a value will never change.  It helps you:
1. Make sure if you add a new constructor that you set the value correctly
2. Avoid adding other code elsewhere that changes the value at a point later than instantiation (which could impact thread safety)

Adding `final` makes it a lot easier to not have to carefully review other accesses of the variable.